invenio_rdm_records/records/api.py

```diff
@@ -80,6 +83,10 @@ class RDMRecord(CommonFieldsMixin, Record):
         delete=False,
     )
 
+    has_draft = HasDraftCheckField(LocalProxy(lambda: RDMDraft))
```

Comment:
I think this can be done with just a lambda and alternatively you can probably just define the draft first, so we don't need the proxy all together.

Reply:
I know this is WIP, so it just a comment to not  forget: We will definitely need some documentation here explaining what this property is used for (like we should have done with ``is_published`` so we don't have to look for how it's used.